drm_*.h are part of kernel DRM API, installed under /usr/include on Linux.
FreeBSD (drm-kmod) doesn't install those anywhere, relying on libdrm copies.
libglvnd replaces libEGL from Mesa, so libdrm dependency needs to be explicit.

--- src/client/configure.json.orig	2020-10-27 08:02:11 UTC
+++ src/client/configure.json
@@ -149,8 +149,7 @@
                     "#endif"
                 ]
             },
-            "libs": "-ldrm",
-            "use": "egl"
+            "use": "egl drm"
         },
         "vulkan-server-buffer": {
             "label": "Vulkan Buffer Sharing",
--- src/compositor/configure.json.orig	2020-10-27 08:02:11 UTC
+++ src/compositor/configure.json
@@ -151,8 +151,7 @@
                     "#endif"
                 ]
             },
-            "libs": "-ldrm",
-            "use": "egl"
+            "use": "egl drm"
         },
         "dmabuf-client-buffer": {
             "label": "Linux Client dma-buf Buffer Sharing",
@@ -176,8 +175,7 @@
                     "return 0;"
                 ]
             },
-            "libs": "-ldrm",
-            "use": "egl"
+            "use": "egl drm"
         },
         "vulkan-server-buffer": {
             "label": "Vulkan Buffer Sharing",
